### DISCUSSION TITLE
How to trigger rule after short power cut

### DISCUSSION THREAD

**Discussion by DerDom** [arendst/Tasmota#22425] [Q&A]
I connected a Shelly with Tasmota to a power line that I can shortly cut by pushing a button. In this case the lamp (connected to the Shelly relay) flickers shortly but the Shelly keeps running (because the power cut is quite short). Is there a way that Tasmota can detect this short power cut and trigger a rule based on this event? 

It should be quite similar to the fast power cycle used to recover devices. Since Tasmota has no problems to detect these cycles, I guess there's a matching trigger. Has someone a clue where to look?

**Comment by sfromis** [comment]
Check the console to see if "anything happens" there with that momentary cut. If not (which your words imply), Tasmota did not "notice" anything, and it would not count for fast power cycle reset.

Tasmota would need to be "notified" about such an event by some hardware feature, if/when it did not lead to the MCU being powered off/on.

To use quick power cycle recovery, my usual recommendation is to have each step be something like 2 seconds, to allow the capacitors time to charge down/up, due to the "risk" that the the residual power can keep the MCU powered for a short time.

**Reply by DerDom** [reply]
Thanks, @sfromis that makes sense.

I will connect L to switch (S1), as well. That should tell quite accurately when power is lost for a short time and I will use Switch1#state=0 to trigger the rule.

**Reply by DerDom** [reply]
Unfortunately, connecting L to S1 didn't work. The console doesn't show a change for Switch1. I guess the residual power keeps S1 powered for a short time, as well!?

Is there a chance that the ADE7953 (part of Shelly Plus 2PM PCB v0.1.9) can recognize a change? Is there a good way to monitor what ADE7953 measures?
